src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -743,4 +744,73 @@ by simp [(Hausdorff_edist_zero_iff_eq_of_closed hs ht).symm, Hausdorff_dist,
          ennreal.to_real_eq_zero_iff, fin]
 
 end --section
+
+section thickening
+
+variables {α : Type u} [pseudo_emetric_space α]
+
+open emetric
+
+/-- The (open) `δ`-thickening `thickening δ E` of a subset `E` in a pseudo emetric space consists
+of those points that are at distance less than `δ` from some point of `E`. -/
+def thickening (δ : ℝ) (E : set α) : set α := {x : α | inf_edist x E < ennreal.of_real δ}
+
+/-- The (open) thickening equals the preimage of an open interval under `inf_edist`. -/
+lemma thickening_eq_preimage_inf_edist (δ : ℝ) (E : set α) :
+  thickening δ E = (λ x, inf_edist x E) ⁻¹' (Iio (ennreal.of_real δ)) := rfl
+
+/-- The (open) thickening is an open set. -/
+lemma is_open_thickening {δ : ℝ} {E : set α} : is_open (thickening δ E) :=
+continuous.is_open_preimage continuous_inf_edist _ is_open_Iio
+
+/-- The (open) thickening of the empty set is empty. -/
+@[simp] lemma thickening_empty (δ : ℝ) : thickening δ (∅ : set α) = ∅ :=
+by { unfold thickening, simp only [set_of_false, inf_edist_empty, not_top_lt], }
```

Comment:
```suggestion
by simp only [thickening, set_of_false, inf_edist_empty, not_top_lt]
```

Reply:
Oh, nice! This trick looks like it can come in handy often. Thank you!